language/move-vm/runtime/src/session.rs

```diff
@@ -55,18 +55,67 @@ impl<'r, 'l, R: RemoteCache> Session<'r, 'l, R> {
         )
     }
 
+    /// Execute a Move script function with the given arguments.
+    ///
+    /// Unlike `execute_function` which is designed for system logic, `execute_script_function` is
+    /// mainly designed to call a script function in an existing module. It similar to
+    /// `execute_script` except that execution of the "script" begins with the specified function
+    ///
+    /// The Move VM MUST return a user error (in other words, an error that's not an invariant
+    /// violation) if
+    ///   - The function does not exist.
+    ///   - The function does not have script visibility.
+    ///   - The signature is not valid for a script. Not all script-visible module functions can
+    ///     be invoked from this entry point. See `script_signature` for the rules.
+    ///   - Type arguments refer to a non-existent type.
+    ///   - Arguments (senders included) fail to deserialize or fail to match the signature of the
+    ///     script function.
+
+    ///
+    /// If any other error occurs during execution, the Move VM MUST propagate that error back to
+    /// the caller.
+    /// Besides, no user input should cause the Move VM to return an invariant violation.
+    ///
+    /// In case an invariant violation occurs, the whole Session should be considered corrupted and
+    /// one shall not proceed with effect generation.
+    pub fn execute_script_function(
+        &mut self,
+        module: &ModuleId,
+        function_name: &IdentStr,
+        ty_args: Vec<TypeTag>,
+        args: Vec<Vec<u8>>,
+        senders: Vec<AccountAddress>,
+        cost_strategy: &mut CostStrategy,
+        log_context: &impl LogContext,
+    ) -> VMResult<()> {
+        self.runtime.execute_script_function(
+            module,
+            function_name,
+            ty_args,
+            args,
+            senders,
+            &mut self.data_cache,
+            cost_strategy,
+            log_context,
+        )
+    }
+
     /// Execute a transaction script.
     ///
-    /// The Move VM MUST return a user error (in other words, an error that's not an invariant violation) if
-    ///   - The script fails to deserialize or verify.
+    /// The Move VM MUST return a user error (in other words, an error that's not an invariant
+    /// violation) if
+    ///   - The script fails to deserialize or verify. Not all expressible signatures are valid.
+    ///     See `script_signature` for the rules.
```

Comment:
Might be hard to understand what `script_signature` refers to here (unless rustdoc will magically create a link to the relevant file)--maybe specify that it's a bytecode verifier pass and given an example of one rule (e.g., no return value)?

Reply:
I'd rather not try to copy out the rules, but I will try to make it more clear that it is a bytecode verifier module